core/git.go

```diff
@@ -96,26 +102,10 @@ func (ref *GitRef) getState(ctx context.Context, bk *buildkit.Client) *llb.State
 		opts = append(opts, llb.AuthHeaderSecret(ref.Repo.AuthHeader.Accessor))
 	}
 
-	useDNS := len(ref.Repo.Services) > 0
-
 	clientMetadata, err := engine.ClientMetadataFromContext(ctx)
-	if err == nil && !useDNS {
-		useDNS = len(clientMetadata.ParentClientIDs) > 0
+	if err != nil {
+		return llb.State{}, err
 	}
 
-	var st llb.State
-	if useDNS {
-		// NB: only configure search domains if we're directly using a service, or
-		// if we're nested beneath another search domain.
-		//
-		// we have to be a bit selective here to avoid breaking Dockerfile builds
-		// that use a Buildkit frontend (# syntax = ...) that doesn't have the
-		// networks API cap.
-		//
-		// TODO: add API cap
```

Comment:
:+1: - I think this comment has been outdated for a while, should be fine to simplify. I might have only even noticed it while testing Bass's Dockerfile `# syntax` support, which has always been janky because of weird Docker <-> Buildkit <-> buildx versioning weirdness anyway (`oci-layout://` source doesn't work for example, so no Container.import). Either way, the c2c networking code doesn't use a networks API cap anymore.

edit: oh wait, this is my own change lol. well I APPROVE